userspace/libsinsp/logger.cpp

```diff
@@ -205,3 +274,58 @@ char* sinsp_logger::format(const char* fmt, ...)
 
 	return m_tbuf;
 }
+
+namespace {
+// All severity strings should be ENCODE_LEN chars long
+const char* sev_levels[] = {
+	"SEV_DEF ",
+	"SEV_FAT ",
+	"SEV_CRI ",
+	"SEV_ERR ",
+	"SEV_WAR ",
+	"SEV_NOT ",
+	"SEV_INF ",
+	"SEV_DEB ",
+	"SEV_TRA "
+};
+
+static_assert(sizeof(sev_levels) == sizeof(*sev_levels) * ((size_t)(sinsp_logger::severity::SEV_MAX) + 1),
+	"severity array must have SEV_MAX+1 elements");
+}
+
+const char *sinsp_logger::encode_severity(const severity sev)
+{
+	const char* ret;
+	auto sev_int = (size_t)sev;
+	if (sev_int > SEV_MAX)
+	{
+		sev_int = 0;
+	}
+
+	ret = sev_levels[sev_int];
+	assert(strlen(ret) == ENCODE_LEN);
+	return ret;
+}
+
+size_t sinsp_logger::decode_severity(const std::string &str, severity& sev)
+{
+	if(str.length() < ENCODE_LEN)
+	{
+		return 0;
+	}
+
+	uint64_t msg_level = *((uint64_t*)str.c_str());
+
+	// we don't really expect "SEV_DEF " messages
+	for(size_t i = 1; i <= SEV_MAX; ++i)
+	{
+		uint64_t level = *((uint64_t*) sev_levels[i]);
+		if(msg_level == level)
```

Comment:
this is fine, but not convinced it buys us much vs just a string compare

Reply:
Even with a hardcoded length of 8, the strncmp isn't optimized out, while the pointer cast (which I changed to a memcpy in the meantime to avoid alignment issues) gets compiled down to `cmp reg, [reg]`:

https://godbolt.org/z/4q2Vhb

Either way, this is nowhere near the hot path (hopefully) so we'll be fine with the strncmp. Thoughts?

Reply:
Settled on using strncmp.